src/Components/Components/src/ComponentsActivitySource.cs

```diff
@@ -0,0 +1,177 @@
+                }
+                if (_httpContext != default)
+                {
+                    activity.AddLink(new ActivityLink(_httpContext));
+                }
+                if (_circuitContext != default)
+                {
+                    activity.AddLink(new ActivityLink(_circuitContext));
+                }
+            }
+
+            activity.DisplayName = $"Route {route ?? "[unknown path]"} -> {componentType ?? "[unknown component]"}";
+            activity.Start();
+            _routeContext = activity.Context;
+        }
+        return activity;
+    }
+
+    public Activity? StartEventActivity(string? componentType, string? methodName, string? attributeName)
+    {
+        var activity = ActivitySource.CreateActivity(OnEventName, ActivityKind.Internal, parentId: null, null, null);
+        if (activity is not null)
+        {
+            if (activity.IsAllDataRequested)
+            {
+                if (_circuitId != null)
+                {
+                    activity.SetTag("aspnetcore.components.circuit.id", _circuitId);
+                }
+                if (componentType != null)
+                {
+                    activity.SetTag("aspnetcore.components.type", componentType);
+                }
+                if (methodName != null)
+                {
+                    activity.SetTag("aspnetcore.components.method", methodName);
+                }
+                if (attributeName != null)
+                {
+                    activity.SetTag("aspnetcore.components.attribute.name", attributeName);
+                }
+                if (_httpContext != default)
+                {
+                    activity.AddLink(new ActivityLink(_httpContext));
+                }
+                if (_circuitContext != default)
+                {
+                    activity.AddLink(new ActivityLink(_circuitContext));
+                }
+                if (_routeContext != default)
+                {
+                    activity.AddLink(new ActivityLink(_routeContext));
+                }
+            }
+
+            activity.DisplayName = $"Event {attributeName ?? "[unknown attribute]"} -> {componentType ?? "[unknown component]"}.{methodName ?? "[unknown method]"}";
+            activity.Start();
+        }
+        return activity;
+    }
+
+    public static void FailEventActivity(Activity? activity, Exception ex)
+    {
+        if (activity != null && !activity.IsStopped)
+        {
+            activity.SetTag("error.type", ex.GetType().FullName);
+            activity.SetStatus(ActivityStatusCode.Error);
+            activity.Stop();
+        }
+    }
+
+    public static async Task CaptureEventStopAsync(Task task, Activity? activity)
+    {
+        try
+        {
+            await task;
+            activity?.Stop();
+        }
+        catch (Exception ex)
+        {
+            FailEventActivity(activity, ex);
+        }
+    }
+}
```

Comment:
This probably needs to go into a different class `ComponentEventActivitySource` (or any other name, RendererEventMetrics or something like that)